clusterous/cluster.py

```diff
@@ -1222,9 +1222,23 @@ def terminate_cluster(self, leave_shared_volume, force_delete_shared_volume):
             raise ClusterException('Cannot connect to AWS')
         
         resource_terminated = False
+        instance_list = self._get_instances(self.cluster_name, connection=conn)
+        
+        # Get shared volume ID
+        shared_volume = None
+        byo_volume = False
+        for i in instance_list:
+            if i.tags.get(defaults.instance_node_type_tag_key) == defaults.controller_name_tag_value:
+                volumes = conn.get_all_volumes(filters={'attachment.instance-id': [i.id]})
+                for v in volumes:
```

Comment:
@balram-ramanathan-sirca It gets only the volumes attached to the "Controller" node. The loop is just to identify which of the two volumes (one is default volume and the other is the attached) is the shared one.